nautobot/ipam/models.py

```diff
@@ -289,14 +333,15 @@ class Meta:
         unique_together = [
             ["vrf", "device"],
             ["vrf", "virtual_machine"],
+            ["vrf", "virtual_device_context"],
             # TODO: desirable in the future, but too strict for 1.x-to-2.0 data migrations,
             #       as multiple "cleanup" VRFs in different cleanup namespaces might be assigned to a single device/VM.
             # ["device", "rd", "name"],
             # ["virtual_machine", "rd", "name"],
```

Comment:
Wonder if we should go ahead and add `virtual_device_context, rd, name` here, or even go ahead and enable that constraint proactively?

Reply:
Added the constraint and added a test case

Reply:
https://github.com/nautobot/nautobot/pull/6936/commits/d0782f498af14906fcf2c02303b001d274e98c95